### PR TITLE
feat: dockerfile support multi arch

### PR DESCRIPTION
**Pull request by AtomicFS**
Preparing for multi-arch support, `arm64` specifically.

I did test the functionality manually on `arm64` machine, so I guess it is fine for now. I definitely still works for `x86` (`amd64`) since the tests are passing.

If there is some issue with `arm64` builds, then I can fix them in PR which will actually use this feature.

### CODE REVIEW

File: docker/coreboot/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 #   Docs: https://docs.docker.com/build/building/multi-stage/
 
+ARG TARGETARCH=amd64
+
 #=============
 # "base" stage with all needed build dependencies
 FROM ubuntu:jammy AS base
 
-ARG TARGETARCH=amd64
+ARG TARGETARCH
```
Comment (MDr164) [inline docker/coreboot/Dockerfile:10]: Can you briefly explain why we define TARGETARCH twice, once with a value (like previously) and now a second time without one?

File: docker/linux/Dockerfile [hunk 2]
```diff
@@ -52,6 +49,12 @@ RUN apt-get update && \
         wget \
         zstd \
     && \
+    if [ "${TARGETARCH}" = 'amd64' ]; then \
```
Comment (MDr164) [inline docker/linux/Dockerfile:52]: Shouldn't we still install different GCC compilers on non amd64 as well? E.g. if one uses arm64 natively but wants to cross compile for amd64 instead.

File: docker/coreboot/Dockerfile (hunk 1 above)
Comment (AtomicFS) [reply docker/coreboot/Dockerfile:10]: Oh yeah, sorry.

It is because of the `ARG` limited [scope](https://docs.docker.com/reference/dockerfile/#scope).

> An ARG instruction goes out of scope at the end of the build stage where it was defined. To use an argument in multiple stages, each stage must include the ARG instruction.

TLDR: `FROM` starts a new scope (a new stage) where the previous `ARG` does not apply anymore. It is possible to use `ARG` without value if the previous `ARG` was declared in the top scope (aka before first `FROM`).

File: docker/linux/Dockerfile (hunk 2 above)
Comment (AtomicFS) [reply docker/linux/Dockerfile:52]: Yes, but I think I included this because I had issues ... but that is a lot of changes and bug-fixes ago, so I will test it again.

Comment (AtomicFS) [reply docker/linux/Dockerfile:52]: Jup, seems to actually work fine. I will remove the `IF` and cleanup it up a bit.

I also noticed that I by accident included the `test_linux.sh` file, I will remove that for now.

Comment (AtomicFS) [reply docker/linux/Dockerfile]: OK, this should be much better now.

**Review by MDr164 — APPROVED** [review]
